### PR TITLE
return a list of multiple matched alias

### PR DESCRIPTION
**Pull request by qingyang-hu**
Make the page display all matched alias rather than redirect to the first one.

I tried to use the full-text search for alias suggestions, but it did not give me what I wanted. As a result, I switched to the regex solution. However, I think it will be better to display all if more than one match. Please let me know your thoughts.

### CODE REVIEW

File: node/mo/pages/[...id].tsx [hunk 1]
```diff
@@ -26,10 +58,14 @@ export async function getServerSideProps(context: any) {
...
     // Search for regexs and replace :)
     let docs = await MoLink.aggregate([{ $match: { $expr: { $regexFind: { input: query, regex: "$alias" } } } }]).exec()
-    if (docs.length > 0) {
+    if (docs.length == 1) {
       let link = docs[0]
       await MoLink.updateOne({ _id: link._id }, { $inc: { n: 1 } }).exec()
       url = query.replace(new RegExp(link.alias), link.link)
+    } else if (docs.length > 1) {
+      return {
+        props: {url: null , links:docs.map(link =>JSON.parse(JSON.stringify({_id:link._id.toString(), alias:link.alias, n:link.n, link:link.link})))},
```
Comment (vector-of-bool) [inline node/mo/pages/[...id].tsx]: `JSON.parse(JSON.stringify(...))` seems odd here?

Comment (ShaneHarvey) [reply node/mo/pages/[...id].tsx]: Can this just be: `props: {url: null , links: docs}`?

Comment (qingyang-hu) [reply node/mo/pages/[...id].tsx]: You are right. I could find a proper JS way to convert it to an object. `{url: null , links: docs}` does not work for me. I'm still training my javascript-fu.

Comment (rozza) [reply node/mo/pages/[...id].tsx]: Just using: `{_id:link._id.toString(), alias:link.alias, n:link.n, link:link.link}` should work here.

**Pull request closed without merging**